sktime/forecasting/compose/_multiplexer.py

```diff
@@ -0,0 +1,188 @@
+
+
+class MultiplexForecaster(
+    _OptionalForecastingHorizonMixin, _SktimeForecaster, _HeterogenousMetaEstimator
+):
+    def __init__(
+        self,
+        forecasters: list,
+        selected_forecaster=None,
+    ):
+
+        """
+        MultiplexForecaster facilitates a framework for performing
+        model selection process over different model classes.
+        It should be used in conjunction with ForecastingGridSearchCV
+        to get full utilization.
+
+        Single use of MultiplexForecaster with forecasters
+        and selected_forecaster parameter specified,
+        works just like the selected component.
+        It does not provide any further use in that case.
+
+        When used with ForecastingGridSearchCV, MultiplexForecaster
+        provides an ability to compare different model class
+        performances with each other, just like a model tournament.
+        When ForecastingGridSearchCV is fitted with a MultiplexForecaster,
+        returned value for the selected_forecaster argument of best_params_
+        attribute of ForecastingGridSearchCV, gives the best
+        performing model class among given models provided in forecasters.
+
+        Parameters
+        ----------
+        forecasters : list
+            List of (forecaster names, forecaster objects)
+            MultiplexForecaster switches between these forecasters
+            objects when used with ForecastingGridSearchCV to
+            find the optimal model
+        selected_forecaster: str
+            An argument to make a selection among forecasters.
+            MultiplexForecaster uses selected_forecaster
+            to choose which component to fit.
+            Important for using with ForecastingGridSearchCV as a
+            hyperparameter.
+
+        Attributes
+        ----------
+        _forecaster : Sktime forecaster
+            forecaster that MultiplexForecaster will currently
+            forecast with.
+        _forecaster_fit_params: dict
+            Fit params for the forecaster that is
+            going to be used by the forecaster in the
+            _forecaster
+        """
+
+        self.forecasters = forecasters
+        self.selected_forecaster = selected_forecaster
+        self._check_components()
+
+        super(MultiplexForecaster, self).__init__()
+
+    def _check_components(self):
+        if not isinstance(self.forecasters, list):
+            raise Exception(
+                "Please provide a list " "for forecasters composed of tuples"
+            )
+
+        for component in self.forecasters:
+            name, forecaster = component
+            if not isinstance(forecaster, _SktimeForecaster):
+                raise Exception(
+                    "Each component has to be an "
+                    "sktime forecaster object. "
+                    "Please check {} with name:{}".format(forecaster, name)
+                )
+
+    def _check_fit_params(self, fit_params):
```

Comment:
Does this establish a new convention or is this something from scikit-learn? What's the expected format of the dictionary?

Reply:
Just read the docstring below, can we not just pass the dictionary with param, value pairs without the estimator name assuming the estimator is the "selected_estimator"?

Reply:
Now also read your comment above on sklearn's pipeline: https://github.com/scikit-learn/scikit-learn/blob/95119c13af77c76e150b753485c662b7c52a41a2/sklearn/pipeline.py#L249 - but still, can we not just pass the ones for the selected estimator?

Reply:
Could we just follow scikit-learn's convention here instead of creating our own?